opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/metric_reader_storage.py

```diff
@@ -51,30 +57,61 @@ def _get_or_init_view_instrument_match(
         # Optimistically get the relevant views for the given instrument. Once set for a given
         # instrument, the mapping will never change
 
-        if instrument in self._view_instrument_match:
-            return self._view_instrument_match[instrument]
+        if instrument in self._instrument_view_instrument_matches:
+            return self._instrument_view_instrument_matches[instrument]
 
         with self._lock:
             # double check if it was set before we held the lock
-            if instrument in self._view_instrument_match:
-                return self._view_instrument_match[instrument]
+            if instrument in self._instrument_view_instrument_matches:
+                return self._instrument_view_instrument_matches[instrument]
 
             # not present, hold the lock and add a new mapping
             view_instrument_matches = []
             for view in self._sdk_config.views:
                 # pylint: disable=protected-access
                 if view._match(instrument):
-                    view_instrument_matches.append(
-                        _ViewInstrumentMatch(
-                            view=view,
-                            instrument=instrument,
-                            sdk_config=self._sdk_config,
-                            instrument_class_aggregation=(
-                                self._instrument_class_aggregation
-                            ),
-                        )
+
+                    new_view_instrument_match = _ViewInstrumentMatch(
+                        view=view,
+                        instrument=instrument,
+                        sdk_config=self._sdk_config,
+                        instrument_class_aggregation=(
+                            self._instrument_class_aggregation
+                        ),
                     )
 
+                    if isinstance(instrument, Asynchronous) and isinstance(
```

Comment:
I thought we agreed on removing this specific check?

Reply:
I think we should check for this, it is mentioned in the spec even if it is an example. I think other situations may cause conflicts that require a view to be discarded, so I have moved this check into a separate method [here](https://github.com/open-telemetry/opentelemetry-python/pull/2608/files#diff-e4ff7543aa8b059f48e9d4b0e3afdcea3baeeee736799473a30a941ef7533869R170), in this way, if more situations that can cause this are detected, we can just add them into this method.